packages/hash/frontend/src/components/hooks/useGetOwnerForEntity.ts

```diff
@@ -0,0 +1,44 @@
+import { Entity, extractOwnedByIdFromEntityId } from "@hashintel/hash-subgraph";
+import { useCallback } from "react";
+import { nilUuid } from "@hashintel/hash-shared/types";
+import { useUsers } from "./useUsers";
+import { useOrgs } from "./useOrgs";
+
+export const useGetOwnerForEntity = () => {
+  /*
+   * This is a simple way of getting all users and orgs to find an entity's owner's name
+   * @todo rethink caching here – users and orgs added since session start won't appear
+   * @todo probably replace this with something like fetching owners individually instead
+   */
+  const { users = [] } = useUsers(true);
+  const { orgs = [] } = useOrgs(true);
+
+  return useCallback(
+    (entity: Entity) => {
+      const ownerUuid = extractOwnedByIdFromEntityId(
+        entity.metadata.editionId.baseId,
+      );
+
+      const owner =
+        // @todo remove this hack when User and Orgs are owned by a real Org
+        ownerUuid === nilUuid
+          ? { orgAccountId: nilUuid, shortname: "system" }
```

Comment:
Has it been changed from "example" to system? I worry about having it inconsistent and then passing it into a route and having it fail. In the code I sent you on slack we special case the system shortname

Reply:
Has what? There's an org called `example-org` but it doesn't own these things (it has a real id)

Reply:
I'm AFK atm but if you search for "example" (with quotes) you'll find it I think. It was there an hour ago

Reply:
(for the root account special cased shortname, which is different from the system org account, and it does not have an associated entity)

Reply:
Both the root account shortname and the default org account use the same environment variable - so this will be checking against `example-org`. I'll update to use this same logic (this from `useGetAccountIdForShortname`).

```
    if (
      shortname === SYSTEM_ACCOUNT_SHORTNAME ||
      shortname === "example" // This is the root account
    ) {
      return nilUuid;
    }
```

Reply:
Done in https://github.com/hashintel/hash/pull/1580/commits/6cb468549715f9fa10ad9bcfe8c3769125815251

Reply:
To flag that the fallback is redundant since `SYSTEM_ACCOUNT_SHORTNAME` itself is always defined, and falls back to `example-org` if the environment variables aren't present